### DISCUSSION TITLE
[Survey] Let us know what parts of `gentoo-haskell` you use!

### DISCUSSION THREAD

**Discussion by hololeap** [gentoo-haskell/gentoo-haskell#1426] [General]
In order to help prevent disruptions for users, I'm putting forth an effort to perform more rigorous build-testing before committing to `master`. To this end, it would be helpful to know what packages people consider essential.

- What packages do you consider critical?
- What packages do you occasionally use but do not consider critical?
- What versions of GHC do you use (latest keyworded, latest unkeyworded, or something specific)?
- What do you use `gentoo-haskell` for and how important is it to your daily work?

---

Currently, the list of critical packages looks approximately like the [list of "leaf packages"](https://github.com/gentoo-haskell/gentoo-haskell/wiki/Leaf-packages-in-main-gentoo-repo) that are pushed to `::gentoo`, but others can be added:

```
app-admin/haskell-updater
app-portage/hackport
app-text/pandoc
dev-haskell/cabal-install
dev-util/shellcheck
dev-vcs/darcs
dev-vcs/git-annex
x11-misc/xmobar
x11-wm/xmonad
x11-wm/xmonad-contrib
```

---

**Please comment and let us know how you use `gentoo-haskell`!**

**Comment by amano-kenji** [comment]
Packages that have patches, such as stack. Patches become outdated over time.

I installed haskell-language-server and cabal from haskell overlay. Without these, I cannot develop haskell programs.

There are choices I can make.

* Build binary packages in a chroot. Install binary packages in the host.
* Use ghcup

**Reply by hololeap** [reply]
Thanks @amano-kenji, I'll add `haskell-language-server` to the list!

**Comment by hololeap** [comment]
I'm adding `ghcup` to the list, since people seem to want it in the overlay.

**Comment by ferki** [comment]
My current list is the following:

```
app-text/pandoc
dev-util/hadolint
dev-util/shellcheck
x11-misc/arbtt
x11-misc/xmobar
x11-wm/xmonad
x11-wm/xmonad-contrib
```

Thanks for maintaining the ebuilds! :heart:

**Comment by l29ah** [comment]
I use the following very important packages daily:
```
app-admin/haskell-updater
x11-wm/xmonad
x11-wm/xmonad-contrib
```
Occasionally:
```
app-text/pandoc
dev-haskell/cabal-install
dev-util/shellcheck
```
I use the latest ghc available that can build xmonad.

**Comment by solpeth** [comment]
My list (excluding packages already mentioned above) is:

- hlint
- hledger and the associated hledger packages (but not -interest or -iadd)
- threadscope
